File: Arrays/Java/Palindromic Array/Palindromic Array.java


/*Complete the Function below*/
class GfG
{
	public static int palinArray(int[] a, int n)
           {
                  //add code here.
                  int y=0;

     for(int i=0;i<n;i++){

         int x=a[i];

         int temp=x;

         int rev=0;

         while(x>0){

             rev=(rev*10)+(x%10);

             x=x/10;

            }

         if(rev==temp)

             y=y+1;

         }

         if(y==n)

             return 1;

         else

             return 0;
           }
}

File: Arrays/CPP/Palindromic Array/Palindromic Array.cpp
class Solution {
    bool palindro(int x){
        int temp = x,sum=0;
        while(temp!=0){
            int modtemp = temp%10;
            sum = sum*10 + modtemp;
            temp = temp/10;
        }
        return sum == x;
    }
public:
    int PalinArray(int a[], int n)
    {
    	for(int i=0;i<n;i++){
    	    if(!palindro(a[i])){
    	        return 0;
    	    }
    	}
    	return 1;
    }
    
};
